R/restrictions.R

```diff
@@ -85,17 +104,25 @@ restrict_nodal_types <- function(model, restriction){
 	}
 
 	# If there are wild cards, spell them out
-	restriction <- lapply(restriction, function(j) unique(unlist(sapply(j, unpack_wildcard))))
+	restriction <- lapply(restriction, function(j) unique(unlist(sapply(j, gbiqq:::unpack_wildcard))))
+
+	# If "keep" specified, reverse meaning of restricted types -- only stipulated types to be kept
+	if(!(action %in% c("remove", "keep"))) stop("action should be either 'remove' to 'keep'")
+	if(action == "keep") for(j in names(restriction)){
+		restriction[[j]] <- nodal_types[[j]][!(nodal_types[[j]] %in% restriction[[j]])]
+	}
```

Comment:
ok I had problems with sapply here because of the different shapes of the data depending on the dimensionality of the input
maybe double check with inputs of different lengths

Reply:
ok let me check that. I didn't see that it could be a problem